### PR TITLE
Added room list logic and added the ability to generate secret santas

### PR DESCRIPTION
**Pull request by andrazbajec**

### CODE REVIEW

File: backend/Model/RoomModel.php [hunk 1]
```diff
@@ -49,7 +48,10 @@ class RoomModel extends AbstractModel
         'RoomID',
         'Rules',
         'MaxAmount',
-        'DateOfExchange'
+        'DateOfExchange',
+        'UserID',
+        'Rules',
```
Comment (tamaragolobic) [inline backend/Model/RoomModel.php:53]: This one is already written 6 rows above

File: backend/Model/RoomModel.php [hunk 2]
```diff
@@ -49,7 +48,10 @@ class RoomModel extends AbstractModel
...
+        'MaxAmount'
```
Comment (tamaragolobic) [inline backend/Model/RoomModel.php:54]: Same as above

File: backend/Model/RoomModel.php [hunk 3]
```diff
@@ -87,15 +89,16 @@ public function createRoom(
         } while ($roomUrlIsNotUnique);
 
         $this->title = $title;
-        $this->shouldJoin = $shouldJoin;
```
Comment (tamaragolobic) [inline backend/Model/RoomModel.php:90]: Why is this one deleted?

File: backend/Model/RoomModel.php [hunk 4]
```diff
@@ -155,7 +158,6 @@ public function save(): RoomModel
         $roomData = [
             'Title' => $this->title,
             'Password' => $this->password ?? null,
-            'ShouldJoin' => $this->shouldJoin ?? null,
```
Comment (tamaragolobic) [inline backend/Model/RoomModel.php:158]: Same as above

File: database/change-20211017-AB.sql [hunk 5]
```diff
@@ -11,10 +11,10 @@ VALUES ('Admin'),
 CREATE TABLE User
 (
     UserID      INT PRIMARY KEY AUTO_INCREMENT,
-    Username    TEXT UNIQUE,
+    Username    VARCHAR(20) UNIQUE,
     Name        TEXT,
     Password    TEXT,
-    Email       TEXT UNIQUE,
+    Email       VARCHAR(50) UNIQUE,
```
Comment (tamaragolobic) [inline database/change-20211017-AB.sql:17]: You didn't rename the file as you should

File: frontend/package.json [hunk 6]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@chakra-ui/icons": "^1.0.16",
```
Comment (tamaragolobic) [inline frontend/package.json:6]: Do we need this icons if we have font awesome?

File: frontend/src/interfaces/RoomInterface.tsx [hunk 7]
```diff
@@ -15,14 +15,47 @@ export interface JoinRoomState {
...
 
 interface User {
     user: string;
+}
+
+export interface RoomListDB {
+    rooms: RoomListDBElement[];
+}
+
+export interface RoomListDBElement {
```
Comment (tamaragolobic) [inline frontend/src/interfaces/RoomInterface.tsx:36]: Do you export this one? If not you don't need export

File: backend/Model/RoomModel.php (hunk 3 above)
Comment (andrazbajec) [reply backend/Model/RoomModel.php:90]: `shouldJoin` does not need to be saved on the `Room` table, it is already evident from the `RoomUser` table.

File: backend/Model/RoomModel.php (hunk 4 above)
Comment (andrazbajec) [reply backend/Model/RoomModel.php:158]: Same anwser as above.

File: database/change-20211017-AB.sql (hunk 5 above)
Comment (andrazbajec) [reply database/change-20211017-AB.sql:17]: Why should the file be renamed? I only updated the type to a limited string as MySQL requires all `UNIQUE` fields to be limited. The file can be updated now, before it's run on production, afterwards new change files will have to be created to update stuff like this.

File: frontend/package.json (hunk 6 above)
Comment (andrazbajec) [reply frontend/package.json:6]: I used chakra ui icons as I don't have font awesome installed. This should stay as is and you'll replace chakra ui icons with font awesome icons and remove the dependency as part of your ticket.

File: backend/Model/RoomModel.php (hunk 1 above)
Comment (andrazbajec) [reply backend/Model/RoomModel.php:53]: It's not among fillable properties?
image

File: backend/Model/RoomModel.php (hunk 2 above)
Comment (andrazbajec) [reply backend/Model/RoomModel.php:54]: This one is not duplicated among fillable properties either.
image

File: frontend/src/interfaces/RoomInterface.tsx (hunk 7 above)
Comment (andrazbajec) [reply frontend/src/interfaces/RoomInterface.tsx:36]: I do export this one, I think what you wanted to ask is if I use it outside this file and I do, `RoomList`, line 19. The export is necessary.

**Pull request merged**